### DISCUSSION TITLE
Capturing fileUri when drag-and-dropping from explorer to webview in a VS Code extension?

### DISCUSSION THREAD

**Discussion by ajthinking** [microsoft/vscode-discussions#1663] [Extension Development QnA] [answered]
I have a webview made with React. I would like to implement a drag-and-drop thing, to allow dropping files from the vs-code explorer onto my webview.

So, in my react code I did a onDrop function, where I call:
```
event.preventDefault();
console.log(event);
```
The thing is, nowhere in the logged event do I find any mention of my dropped file uri (./todos.json or similar). How can I access that?

Comparing with dropping files from "finder" ie macs folders which works great. For those case I can find data in:

event.dataTransfer.files
But that one (and even event.nativeEvent.dataTransfer) is empty when dropping from vs-code's explorer.

Any ideas? What I can try here?

**Comment by gjsjohnmurray** [comment]
Based on what I read in https://github.com/microsoft/vscode/issues/182449 do you see any difference if you hold Shift down?

**Reply by ajthinking** [reply]
Yes - I need to hold SHIFT to allow dropping into my react app. Otherwise it will just open a new standard editor with the dropped file. When holding SHIFT and dropping I see my event logged, so that part is working. But the event contains no relevant data 🤔 

Let me read up on that issue! Might help to get unstuck!
Thank you!

**Reply by ajthinking** [reply, marked as answer]
Found solution to my problem in the demo repo of linked thread! Thank you @gjsjohnmurray  🥳 

I needed to call getData to reveal the uri, like so:
```js
  const allDropVariations = JSON.stringify({
    'dataTransfer.types': Array.from(event.dataTransfer.types),
    'dataTransfer.getData(text/uri-list)': event.dataTransfer.getData('text/uri-list'),
    'dataTransfer.getData(text/plain)': event.dataTransfer.getData('text/plain'),
    'dataTransfer.files.0.name': event.dataTransfer.files.item(0)?.name,
  }, null, 2);

  console.log('Dropped:', allDropVariations);
```
